### PR TITLE
Player Kick Method Choice

### LINKED ISSUE #1281: [Request]: Choose Kick in Reactions
### Problem

_No response_

### Solution

Add a List of Kick's after checking the box "Kick Player", 
Make it that you can select only one.
List : 
Null Function Kick.
End Session Kick.
Desync Kick.
OOM Kick.
Bail Kick.

### Reason

I sometimes wanna use others kicks in reactions

### Additional context

Oh and If a kick Doesn't Work Out then Use the other one

### PR DESCRIPTION
**Pull request by Rxann**
I need help with this. The menu items for end and null function kick do not work. I don't even know why in the hell they don't. This seems to be what I need help with.

closes #1281

### CODE REVIEW

File: src/views/settings/view_reaction_settings.cpp [hunk 1]
```diff
@@ -102,6 +104,13 @@ namespace big
 		draw_reaction(g.reactions.gamer_instruction_kick);
 		draw_interloper_reaction(g.reactions.breakup_others);
 
+		components::title("Reaction Kick Method");
+		components::sub_title("Kick Option:");
+		ImGui::Checkbox("Default", &g.settings.default_reaction_kick);
+		ImGui::Checkbox("Breakup Kick", &g.settings.breakup_reaction_kick);
+		ImGui::Checkbox("End Session Kick", &g.settings.end_reaction_kick);
```
Comment (maybegreat48) [inline src/views/settings/view_reaction_settings.cpp]: Try `"End Session Kick##kick_method"` and `"Null Function Kick##kick_method"`

File: src/backend/reactions/reaction.cpp [hunk 2]
```diff
@@ -39,10 +41,37 @@ namespace big
...
+				}
+				else if (g.settings.end_reaction_kick)
+				{
+					dynamic_cast<player_command*>(command::get(RAGE_JOAAT("endkick")))->call(player, {});
+				}
+				else if (g.settings.nf_reaction_kick)
+				{
+					dynamic_cast<player_command*>(command::get(RAGE_JOAAT("nfkick")))->call(player, {});
+					script::get_current()->yield(700ms);
+					if (player->is_valid())
```
Comment (maybegreat48) [inline src/backend/reactions/reaction.cpp:68]: Null function kick should take about 15 seconds to work if they haven't blocked it so you would use the breakup kick anyway. End kick is also blocked by most menus

File: src/views/settings/view_reaction_settings.cpp (hunk 1 above)
Comment (Rxann) [reply src/views/settings/view_reaction_settings.cpp]: Should Default and Breakup Kick have these also?

File: src/backend/reactions/reaction.cpp (hunk 2 above)
Comment (Rxann) [reply src/backend/reactions/reaction.cpp:68]: I have changed the script yield time to 15seconds, and do you know another kick I could use as a replacement for end kick? or if I even should have another kick?

**Comment by Rxann** [comment]
I think there may be an issue with the global or checkbox because when the menu is opened none of the option checkboxes are checked.

File: src/backend/reactions/reaction.cpp (hunk 2 above)
Comment (maybegreat48) [reply src/backend/reactions/reaction.cpp:68]: You can try the bail kick, but literally everything has protections for that

Comment (Rxann) [reply src/backend/reactions/reaction.cpp:68]: Okay, I think we can just leave it with end kick

File: src/views/settings/view_reaction_settings.cpp (hunk 1 above)
Comment (maybegreat48) [reply src/views/settings/view_reaction_settings.cpp]: You don't need it if it works

Comment (Rxann) [reply src/views/settings/view_reaction_settings.cpp]: @maybegreat48 the default option isn't checked when I open it

**Comment by Rxann** [comment]
Could anyone review this code because I believe that the default option isn't checked by default when opening the reactions.

**Comment by Yimura** [comment]
> Could anyone review this code because I believe that the default option isn't checked by default when opening the reactions.

As I said [here](https://github.com/YimMenu/YimMenu/discussions/1286) I will not be spending time on "enhancement" PR's/PR's that add new features to the menu.
I'm completely focussing on improving the code base and fixing bugs.

**Comment by Rxann** [comment]
> > Could anyone review this code because I believe that the default option isn't checked by default when opening the reactions.
> 
> As I said [here](https://github.com/YimMenu/YimMenu/discussions/1286) I will not be spending time on "enhancement" PR's/PR's that add new features to the menu. I'm completely focussing on improving the code base and fixing bugs.

Ah okay I see. I will try to debug it myself

**Comment by Rxann** [comment]
Alright, this feature works, I just needed to delete my settings file 😆

**Comment by Rxann** [comment]
It should be good to merge assuming no major bugs have been found

**Comment by Rxann** [comment]
this is pretty stale, can be closed

**Comment by gir489returns** [comment]
So then close it.

**Pull request closed without merging**